tests/Sinch.Tests/Core/UrlResolverTests.cs

```diff
@@ -314,5 +315,60 @@ public void ResolveSmsServicePlanIdUrl(SmsServicePlanIdRegion smsServicePlanIdRe
                 : new Uri(apiUrlOverrides.SmsUrl);
             smsUrl.Should().BeEquivalentTo(expectedUrl);
         }
+
+
+        public static IEnumerable<object[]> MailgunUrlData => new List<object[]>
+        {
+            new object[]
+            {
+                MailgunRegion.Us,
+                null,
+            },
+            new object[]
+            {
+                MailgunRegion.Eu,
+                null,
+            },
+            new object[]
+            {
+                MailgunRegion.Eu,
+                new ApiUrlOverrides()
+                {
+                    MailgunUrl = null
+                }
+            },
+            new object[]
+            {
+                MailgunRegion.Eu,
+                new ApiUrlOverrides()
+                {
+                    MailgunUrl = "https://my-mailgun-proxy.net"
+                }
+            },
+        };
+
+        [Theory]
+        [MemberData(nameof(MailgunUrlData))]
+        public void ResolveMailgunUrl(MailgunRegion region, ApiUrlOverrides apiUrlOverrides)
```

Comment:
There shouldn't be any logic in this test function: here you are recoding the logic from the UrlResolver. It would be better to provide the expected value as part of the test data. And also adding a test name to improve the readability, such as:
```
public static IEnumerable<object[]> ResolveMailgunUrlTestCases => new List<object[]>
{
    // Format: TestName, Region, Override, ExpectedUrl
    new object[] { "US Region Default URL", MailgunRegion.Us, null, "https://api.mailgun.net" },
    new object[] { "EU Region Default URL", MailgunRegion.Eu, null, "https://api.eu.mailgun.net" },
    new object[] { "EU Region with Null Override", MailgunRegion.Eu, new ApiUrlOverrides { MailgunUrl = null }, "https://api.eu.mailgun.net" },
    new object[] { "EU Region with Custom Override", MailgunRegion.Eu, new ApiUrlOverrides { MailgunUrl = "https://my-mailgun-proxy.net" }, "https://my-mailgun-proxy.net" },
};
```

The test would be more straightforward:
```
[Theory]
[MemberData(nameof(ResolveMailgunUrlTestCases))]
public void ResolveMailgunUrl(string testName, MailgunRegion region, ApiUrlOverrides apiUrlOverrides, string expectedUrl)
{
    var actual = new UrlResolver(apiUrlOverrides).ResolveMailgunUrl(region);
    actual.Should().BeEquivalentTo(new Uri(expectedUrl));
}
```

And in order to test all the branches from the UrlResolver method, you can consider adding another test for an unexpected Region value.

Reply:
even with `ResolveMailgunUrl((MailgunRegion)(-1))` ?
I thought this was why you added a `default` clause in the switch statement in the ResolveMailgunUrl method